main_test.go

```diff
@@ -80,6 +82,16 @@ func TestInsertionHappyPath(t *testing.T) {
 	if response.StatusCode != http.StatusOK {
 		t.Fatalf("Expected status code %d, got %d", http.StatusOK, response.StatusCode)
 	}
+
+	responseBody, err := io.ReadAll(response.Body)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var ir prover.InsertionResponse
+	if err = json.Unmarshal(responseBody, &ir); err != nil {
+		t.Fatal(err)
+	}
```

Comment:
would be good if this test also verified if the proof can be verified after umarshalling

Reply:
VK should be available from the proving system, not the response. You don't need the witness, just the public variables (i.e. you can construct a public-only witness easily)